component/pdp/component.go

```diff
@@ -0,0 +1,180 @@
+package pdp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"slices"
+
+	"github.com/nuts-foundation/nuts-knooppunt/component"
+	"github.com/nuts-foundation/nuts-knooppunt/component/mitz"
+	"github.com/nuts-foundation/nuts-knooppunt/component/mitz/xacml"
+)
+
+type Config struct {
+	Enabled bool
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Enabled: true,
+	}
+}
+
+var _ component.Lifecycle = (*Component)(nil)
+
+type Component struct {
+	Config Config
+	Mitz   *mitz.Component
+}
+
+// New creates an instance of the pdp component, which provides a simple policy decision endpoint.
+func New(config Config, mitzcomp *mitz.Component) (*Component, error) {
+	return &Component{
+		Config: config,
+		Mitz:   mitzcomp,
+	}, nil
+}
+
+func (c Component) Start() error {
+	// Nothing to do
+	return nil
+}
+
+func (c Component) Stop(ctx context.Context) error {
+	// Nothing to do
+	return nil
+}
+
+func (c Component) RegisterHttpHandlers(publicMux *http.ServeMux, internalMux *http.ServeMux) {
+	internalMux.HandleFunc("POST /pdp", http.HandlerFunc(c.HandleMainPolicy))
+	internalMux.HandleFunc("POST /pdp/v1/data/{package}/{rule}", http.HandlerFunc(c.HandlePolicy))
+}
+
+type MainPolicyInput struct {
+	Method                           string   `json:"method"`
+	Path                             []string `json:"path"`
+	PatientBSN                       string   `json:"patient_bsn"`
+	RequestingUziRoleCode            string   `json:"requesting_uzi_role_code"`
+	RequestingPractitionerIdentifier string   `json:"requesting_practitioner_identifier"`
+	RequestingOrganizationUra        string   `json:"requesting_organization_ura"`
+	RequestingFacilityType           string   `json:"requesting_facility_type"`
+	DataHolderOrganizationUra        string   `json:"data_holder_organization_ura"`
+	DataHolderFacilityType           string   `json:"data_holder_facility_type"`
+	PurposeOfUse                     string   `json:"purpose_of_use"`
+}
```

Comment:
We also send `ResourceType` and `ResourceId` due to OPA. We don't use them, but it might be nice to add them to the struct as they are part of its input.